File: source/XeroApi.Tests/Stubs/StubIntegrationProxy.cs
using System;
using System.IO;
using System.Text;
using XeroApi.Integration;
using XeroApi.Linq;
using XeroApi.Model;

namespace XeroApi.Tests.Stubs
{
    internal class StubIntegrationProxy : IIntegrationProxy
    {
        public string FindElements(IApiQueryDescription apiQueryDescription)
        {
            LastQueryDescription = apiQueryDescription as LinqQueryDescription;
            return GenerateSampleResponseXml(apiQueryDescription.ElementName);
        }

        public string FindAttachments(string endpointName, string itemId)
        {
            throw new NotImplementedException();
        }

        public byte[] FindOne(string endpointName, string itemId, string acceptMimeType)
        {
            return Encoding.UTF8.GetBytes(GenerateSampleResponseXml(endpointName));
        }

        public Stream FindOneAttachment(string endpointName, string itemId, string attachmentIdOrFileName)
        {
            throw new NotImplementedException();
        }

        public string GetElement(string endpointName, string itemId)
        {
            LastEndpointName = endpointName;
            return GenerateSampleResponseXml(endpointName);
        }

        public string UpdateOrCreateElements(string endpointName, string body)
        {
            LastEndpointName = endpointName;
            return GenerateSampleResponseXml(endpointName);
        }

        public string UpdateOrCreateAttachment(string endpointName, string itemId, Attachment attachment)
        {
            throw new NotImplementedException();
        }

        public string CreateElements(string endpointName, string body)
        {
            return GenerateSampleResponseXml(endpointName);
        }

        public string CreateAttachment(string endpointName, string itemId, Attachment attachment)
        {
            throw new NotImplementedException();
        }

        public string ApplyAllocation(CreditNote creditNote, string requestXml)
        {
            throw new NotImplementedException();
        }

        public LinqQueryDescription LastQueryDescription
        {
            get;
            private set;
        }

        public string LastEndpointName
        {
            get; 
            private set; 
        }

        private static string GenerateSampleResponseXml(string elementName)
        {
            return @"
<Response>
    <Id>" + Guid.NewGuid() + @"</Id>
    <Status>OK</Status>
    <ProviderName>NullIntegrationProxy</ProviderName>
    <DateTimeUTC>" + DateTime.UtcNow.ToString("s") + @"</DateTimeUTC><" + elementName + "s" + @" />
    <" + elementName + @"s>
        <" + elementName + @"/>
    </" +elementName+@"s>
</Response>";
        }
    }

}

File: source/XeroApi.VBTests/Stubs/StubIntegrationProxy.vb
Imports System.IO
Imports XeroApi.Integration
Imports XeroApi.Model

Public Class StubIntegrationProxy
    Implements IIntegrationProxy

    
    Public Function FindElements(apiQueryDescription As IApiQueryDescription) As String Implements IIntegrationProxy.FindElements
        LastQueryDescription = apiQueryDescription
        Return GenerateSampleResponseXml(apiQueryDescription.ElementName)
    End Function


    Public Function FindAttachments(endpointName As String, itemId As String) As String Implements IIntegrationProxy.FindAttachments
        Return GenerateSampleResponseXml(endpointName)
    End Function


    Public Function FindOne(endpointName As String, itemId As String, acceptMimeType As String) As Byte() Implements IIntegrationProxy.FindOne
        Return Nothing
    End Function


    Public Function FindOneAttachment(endpointName As String, itemId As String, attachmentIdOrFileName As String) As Stream Implements IIntegrationProxy.FindOneAttachment
        Return Stream.Null
    End Function


    Public Function UpdateOrCreateElements(endpointName As String, body As String) As String Implements IIntegrationProxy.UpdateOrCreateElements
        Return GenerateSampleResponseXml(endpointName)
    End Function


    Public Function UpdateOrCreateAttachment(endpointName As String, itemId As String, attachment As Attachment) As String Implements IIntegrationProxy.UpdateOrCreateAttachment
        Return GenerateSampleResponseXml(endpointName)
    End Function


    Public Function CreateElements(endpointName As String, body As String) As String Implements IIntegrationProxy.CreateElements
        Return GenerateSampleResponseXml(endpointName)
    End Function


    Public Function CreateAttachment(endpointName As String, itemId As String, attachment As Attachment) As String Implements IIntegrationProxy.CreateAttachment
        Return GenerateSampleResponseXml(endpointName)
    End Function

    Private Function GenerateSampleResponseXml(elementName As String) As String

        Return "<Response><Id>" & Guid.NewGuid().ToString & "</Id><Status>OK</Status><ProviderName>NullIntegrationProxy</ProviderName><DateTimeUTC>" & DateTime.UtcNow.ToString("s") & "</DateTimeUTC><" & elementName & "s" & " /></Response>"

    End Function

    Public LastQueryDescription As IApiQueryDescription = Nothing

    Public Function ApplyAllocation(creditNote As CreditNote, xml As String) As String Implements IIntegrationProxy.ApplyAllocation
        Return String.Empty
    End Function

End Class
